src/packaging/kpatch/commit_to_overlay.sh

```diff
@@ -0,0 +1,459 @@
+    rm -fv ${commit_file}
+  fi
+
+  unset -f CHANGE_LOG DEBUG GITDATE GITAUTHOR LINUX_SRC
+}
+
+###############################################################################
+# main()
+trap _cleanup EXIT
+test "$#" -lt 5 && print_usage
+
+process_args $@
+print_config
+checkout_source_branch
+$PROCESS_INPUT_FX
+
+echo # Intentional blank line
+
+# Store linux repo location as an absolute path in the current directory
+LINUX_SRC=${RUN_DIR}/$(mktemp -d vdo-kernel-XXXXXX)
+VDO_TREE=${RUN_DIR}/$(mktemp -d vdo-raw-XXXXXX)
+
+# Shallow clone the specified dm-linux repo branch into LINUX_SRC
+echo "Cloning kernel repo $(get_repo_name ${KERNEL_REPO_URL})/${KERNEL_BRANCH}"
+git clone --branch ${KERNEL_BRANCH} --depth 1 ${KERNEL_REPO_URL} ${LINUX_SRC}
+cd $LINUX_SRC
+git checkout -b ${OVERLAY_BRANCH} ${KERNEL_BRANCH}
+
+# Need to detect that our repo is x number of commits ahead of dm-vdo/vdo-devel
+# Clone the current repo as ${VDO_TREE}? Is VDO_TREE even needed? Aren't we working from the clone when this script is called? Need to make a copy?
+echo -en "\nCloning VDO tree $(get_repo_name ${SOURCE_REPO_URL})\n"
+git clone $SOURCE_REPO_URL ${VDO_TREE}
+
+# Loop over each commit in COMMIT_SHAS and overlay it on LINUX_SRC
+for change in ${COMMIT_SHAS[@]}; do
+  echo -en "\nProcessing changes from commit $change\n"
+
+  # Remove existing dm-vdo in case we have any deletes.
+  # XXX: The kpatch overlay process doesn't handle deletes at the moment, which
+  #      is a bug that should get handled in the future by the Makefile process
+  #      there, but it's going to get dealt with here for now.
+  if [ -d drivers/md/dm-vdo ]; then
+    rm -rf drivers/md/dm-vdo
+  fi
+
+  cd ${VDO_TREE}
+
+  # Clean up from the last build, if necessary
+  if [[ "${change}" != "${COMMIT_SHAS[0]}" ]]; then
+    echo -en "Cleaning up from the last build... "
+    git clean -fdxq
+    echo "Done"
+  fi
+
+  # Check out the new version to build and post - do we need to do this - it puts us in a detached HEAD state, so commits will not be retained
+  #sudo git config --system advice.detachedHead "false"
+  echo "Checking out the change to build and post it"
+  git checkout ${change}
+
+  # Capture the commit message from the change
+  echo "Creating the initial commit file using the commit message from the change"
+  commit_file=$(mktemp /tmp/overlay_commit_file.XXXXX)
+  git log --format=%B -n 1 ${change} > ${commit_file}
+
+  # Capture the source commit hash into the commit message as well
+  echo "Original commit: ${change}" >> ${commit_file}
+
+  # Capture authorship information for use later
+  commit_author_email=$(git log --format=%aE -1 ${change})
+  commit_author_name=$(git log --format=%aN -1 ${change})
+  commit_date=$(git log --format=%ad -1 ${change})
+  gitauthor="--author \"${commit_author_name} <${commit_author_email}>\""
+  gitdate="--date \"${commit_date}\""
+
+  # Build the tree as normal
+  build_log_file=$(mktemp /tmp/overlay_build_log.XXXXX)
```

Comment:
Again, should the logs go in ${RUN_DIR}?

Reply:
Potentially. I chose /tmp for the reasons mentioned above. Although I can change that if you think it is better in ${RUN_DIR}.

Reply:
I see the error message directly names the build_log file, which allays my concern about finding it. So having it live in /tmp is fine.